Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -165,6 +165,13 @@ class Frontend {
   /// Read and prepare the file at the given path for processing, optionally synchronizing
   /// diagnostic output.
   private func openAndPrepareFile(at url: URL) -> FileToProcess? {
+    guard url.lastPathComponent != IgnoreFile.standardFileName else {
```

Comment:
This way of reporting errors from invalid `.swift-format-ignore` files seems a little roundabout to me. What do you think about gathering errors in `FileIterator` inside a `var diagnostics: [Diagnostic]` member. Then inside `Frontend.processURLs` where we construct the `FileIterator`, we can consult the diagnostics produced by it and report them. That way we should also be able to test that `FileIterator` correctly emits these diagnostics.